exercise/submission_models.py

```diff
@@ -829,6 +829,22 @@ def is_grader_stable(self):
         return not (total_retries and total_retries > settings.GRADER_STABLE_THRESHOLD)
 
 
+    def get_exercise_names_if_grader_is_unstable(self, instance):
+        total_retries_per_exercise = self.values(
+            'submission__exercise__name',
+        ).filter(
+            submission__exercise__course_module__course_instance=instance.id
+        ).annotate(
+            num_retries=models.Sum('num_retries')
+        )
+        total_retries = sum(entry['num_retries'] for entry in total_retries_per_exercise)
+        # Check if the grader can be considered unstable on this course instance
+        if total_retries > settings.GRADER_STABLE_THRESHOLD:
+            exercises = ", ".join(f"'{entry['submission__exercise__name']}'" for entry in total_retries_per_exercise)
```

Comment:
If the course has 100 exercises and all of them are not working, then is the list of exercise names going to grow too much? If all of them are rendered in the alert, it becomes a mess. What if the database query is limited to take only the 10 first exercises? Maybe the user does not need an exact list of all the exercises when there are clearly many problematic exercises.

Reply:
Done! It now only queries for the first 10 exercises with the most retries!